pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -132,6 +147,102 @@ func (d *depsCollector) populatePurls(ctx context.Context, docChannel chan<- *pr
 	return nil
 }
 
+// getAllDependencies gets all the dependencies for the purls provided in a concurrent manner.
+func (d *depsCollector) getAllDependencies(ctx context.Context, purls []datasource.Source) error {
+	// channels to signal when the project and version info have been fetched
+	projectDone := make(chan bool)
+	versionDone := make(chan bool)
+
+	// channels to send the inputs to the goroutines
+	projectChan := make(chan *pb.ProjectKey)
+	versionChan := make(chan *pb.VersionKey)
+	logger := logging.FromContext(ctx)
+
+	// these goroutines will be used to version info concurrently
+	go func() {
+		// this go routine has to be before the next go routine as it will be pushing into the project channel
+		// for each version that is fetched from the version channel it will check if the project has to be fetched
+		d.versions = d.getVersions(ctx, versionChan, projectChan) // the results are the stored in the versions map
+		versionDone <- true
+	}()
+
+	// these goroutines will be used to fetch the projects concurrently
+	go func() {
+		// this sets up the goroutine to fetch the projects concurrently for each input
+		d.projectInfoMap = d.getProjects(ctx, projectChan) // the results are the stored in the projectInfoMap map
+		// posts to the projectDone channel to signal that all projects have been fetched
+		projectDone <- true
+	}()
+
+	// TODO: Concurrently fetch the dependencies for each purl
+	for _, p := range purls {
+		purl := p.Value
+		packageInput, err := helpers.PurlToPkg(purl)
+		if err != nil {
+			logger.Infof("failed to parse purl to pkg: %s", purl)
+			return nil
+		}
+
+		// if version is not specified, cannot obtain accurate information from deps.dev. Log as info and skip the purl.
+		if *packageInput.Version == "" {
+			logger.Infof("purl does not contain version, skipping deps.dev query: %s", purl)
+			return nil
+		}
+
+		// Make an RPC Request. The returned result is a stream of
+		// DependenciesResponse structs.
+		versionKey, err := getVersionKey(packageInput.Type, packageInput.Namespace, packageInput.Name, packageInput.Version)
+		if err != nil {
+			logger.Infof("failed to getVersionKey with the following error: %v", err)
+			return nil
+		}
+		// send the version key to the version channel
+		versionChan <- versionKey
+
+		dependenciesReq := &pb.GetDependenciesRequest{
+			VersionKey: versionKey,
+		}
+
+		deps, err := d.client.GetDependencies(ctx, dependenciesReq)
+		if err != nil {
+			logger.Debugf("failed to get dependencies %v", err)
```

Comment:
Errorf instead?

Reply:
Same as the others, changed it to Debugf.